src/VisualStudio/Core/Def/Implementation/LanguageClient/RazorInProcLanguageClient.cs

```diff
@@ -25,8 +25,16 @@ namespace Microsoft.CodeAnalysis.ExternalAccess.Razor.Lsp
     /// TODO - This can be removed once C# is using LSP for diagnostics.
     /// https://github.com/dotnet/roslyn/issues/42630
     /// </summary>
+    /// <remarks>
+    /// This specifies RunOnHost because in LiveShare we don't want this to activate on the guest instance
+    /// because LiveShare drops the ClientName when it mirrors guest clients, so this client ends up being
+    /// activated solely by its content type, which means it receives requests for normal .cs and .vb files
+    /// even for non-razor projects, which then of course fails because it gets text sync info for documents
+    /// it doesn't know about.
+    /// </remarks>
     [ContentType(ContentTypeNames.CSharpContentType)]
     [ClientName(ClientName)]
+    [RunOnContext(RunningContext.RunOnHost)]
```

Comment:
This should be inferred. I'm really really surprised you have to add this. I have a suspicion that something else may have gone wrong here in our dependencies. I'd highly suggest reaching out to Miguel Lopez to validate that you need this.

Reply:
Inferred from what? The presence of client name?

I still have been unable to validate this unfortunately, so I'm not banking on this just yet :)

Reply:
"Inferred" may have been the wrong word choice. This should be the default for language clients

Reply:
@NTaylorMullen I think RunOnHost means that the server is not mirrored, which is not the default (by default the server is mirrored to the guest)
https://devdiv.visualstudio.com/DevDiv/_git/VSLanguageServerClient?path=%2Fsrc%2Fproduct%2FLanguageServer.Client.Definition%2FRunOnContextAttribute.cs&version=GBdevelop&line=16&lineEnd=17&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents